File: LeetCode/557. Reverse Words in a String III.js
// https://leetcode.com/problems/reverse-words-in-a-string-iii/

const { test } = require('../test');

const reverseWords = (str) => {
  let temp = '';

  let result = '';

  for (let i = 0; i < str.length; i++) {
    if (str[i] === ' ') {
      result += temp + ' ';
      temp = '';

      continue;
    }

    temp = str[i] + temp;
  }

  result += temp;

  return result;
};

test(reverseWords, [
  {
    input: ["Let's take LeetCode contest"],
    expected: "s'teL ekat edoCteeL tsetnoc",
  },
  {
    input: ['Mr Ding'],
    expected: 'rM gniD',
  },
]);

File: LeetCode/557. Reverse Words in a String III.py
# https://leetcode.com/problems/reverse-words-in-a-string-iii/

from test import test


class Solution:
    # def reverseWords(self, text: str) -> str:
    #     return " ".join(map(lambda word: "".join(reversed(word)), text.split(" ")))
    def reverseWords(self, text: str) -> str:
        return " ".join(word[::-1] for word in text.split())


solution = Solution()


test(
    solution.reverseWords,
    [
        {
            "input": ["Let's take LeetCode contest"],
            "expected": "s'teL ekat edoCteeL tsetnoc",
        },
        {
            "input": ["Mr Ding"],
            "expected": "rM gniD",
        },
    ],
)
